### PR TITLE
Heidenbzr async parallel dictionary

### PR DESCRIPTION
**Pull request by HeidenBZR**

### CODE REVIEW

File: OpenUtau.Core/Render/RenderEngine.cs [hunk 1]
```diff
@@ -78,19 +78,20 @@ public class Progress {
...
-                    MaxDegreeOfParallelism = threads
-                }, body: item => {
-                    if (source.Token.IsCancellationRequested) {
-                        return;
-                    }
-                    item.progress = progress;
-                    Resample(item);
-                });
-                progress.Clear();
+                if (items.Count > 0) {
```
Comment (HeidenBZR) [inline OpenUtau.Core/Render/RenderEngine.cs:82]: "phonemizer is initializing" was overlapped by this "Clear" even if there was no render items

File: OpenUtau/ViewModels/NotesViewModel.cs [hunk 2]
```diff
@@ -509,6 +509,10 @@ public class NotesViewModel : ViewModelBase, ICmdSubscriber {
                     if (focusNote.part == Part) {
                         FocusNote(focusNote.note);
                     }
+                } else if (cmd is ValidateProjectNotification) {
+                    Project.Validate();
```
Comment (stakira) [inline OpenUtau/ViewModels/NotesViewModel.cs:513]: Please keep Project.Validate() call inside DocManager.

File: OpenUtau.Core/Render/RenderEngine.cs (hunk 1 above)
Comment (stakira) [reply OpenUtau.Core/Render/RenderEngine.cs:82]: Please prefer returning early, to reduce nesting i.e.
```
if (item.Count == 0) {
  return;
}
```

File: OpenUtau.Core/Api/G2pDictionary.cs [hunk 3]
```diff
@@ -30,13 +30,20 @@ class TrieNode {
...
                         .ToList()
                         .ForEach(parts => builder.AddSymbol(parts[0], parts[1]));
-                    Core.Api.Resources.cmudict_0_7b.Split('\n')
-                        .Where(line => !line.StartsWith(";;;"))
-                        .Select(line => line.Trim().ToLowerInvariant())
-                        .Select(line => line.Split(new string[] { "  " }, StringSplitOptions.None))
-                        .Where(parts => parts.Length == 2)
-                        .ToList()
-                        .ForEach(parts => builder.AddEntry(parts[0], parts[1].Split().Select(symbol => RemoveTailDigits(symbol))));
+                    Core.Api.Resources.cmudict_0_7b.Split('\n').AsParallel().ForAll(line => {
```
Comment (stakira) [inline OpenUtau.Core/Api/G2pDictionary.cs:33]: Parallelizing a lot of small tasks is overkill and may actually slow things down.
A single thread as the original is not too bad after you made it async. The builder is not thread-safe either.

**Pull request merged**